src/window/window_wrapper.rs

```diff
@@ -112,6 +117,40 @@ impl WinitWindowWrapper {
 
         let ime_enabled = { SETTINGS.get::<KeyboardSettings>().ime };
 
+        #[cfg(target_os = "macos")]
+        let mut ks = SETTINGS.get::<KeyboardSettings>();
+        #[cfg(target_os = "macos")]
+        {
+            let window = windowed_context.window();
+            if ks.macos_option_key_is_meta.0 == OptionAsAlt::None && ks.macos_alt_is_meta {
+                // default values are 'OptionAsAlt::None' and 'false', respectively, so if we
+                // arrive here it's most likely because the user set only the legacy setting
+                // (alt_is_meta) and not the new one (option_key_is_meta), so:
+                ks.macos_option_key_is_meta.0 = OptionAsAlt::Both;
+                SETTINGS.set::<KeyboardSettings>(&ks);
+                // This is probably not the best way to do this, because by the time we call
+                // SETTINGS.get() above, we've lost the ability to be affected by the order in
+                // which settings arrive from nvim (so we don't really know whether
+                // 'alt_is_meta' is overriding 'option_key_is_meta' or vice-versa).
+                //
+                // Ideally, we would have a custom entry for each of
+                // "neovide_input_macos_alt_is_meta" and
+                // "neovide_input_macos_option_key_is_meta" in SETTINGS.listeners, and those
+                // update functions would directly call winit's window.set_option_as_alt() as
+                // we do here (since, after receiving the update from nvim and updating winit,
+                // window.option_as_alt() is arguably the source of truth for this setting).
+                // Repeat this for 'ime', and then we wouldn't need the KeyboardSettings
+                // struct at all, and we wouldn't need the duplicate values stored in
+                // WinitWindowWrapper::{macos_option_is_meta, ime_enabled}.
+                //
+                // Come to think of it... do we really need to synchronize all settings once
+                // per video frame? It seems like we'd generally only need a synchronization
+                // once per "setting_changed" notification from nvim (and then only for one
+                // setting, not all of them)...?
+            }
```

Comment:
I think it would be cleaner if this was moved into it's own `get_option_as_alt_setting` function. 

And then called at synchronise settings, that way it can also be updated on the fly, which I don't think is possible with your current implementation.

I also don't think you need to call it here, you can just set it to `None`, and let the synchronise settings take care of actually setting the value.

I agree that using events would be a better approach, but our settings system currently does not support that. So, I guess we will have to live with `synchronise_settings`  approach for now. 

I don't believe that the order matters, since we want to deprecate the old `macos_option_is_meta` option. So it's really a user fault if both options are set, so if the new one always overrides, I think that's ok.

Reply:
We do have a settings listener now. So I think that should clean this up.